tools/integration_tests/monitoring/prom_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/mounting"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/util"
+	promclient "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// A directory containing outputs created by build_gcsfuse.
+var gBuildDir string
+
+const (
+	testBucket    = "gcsfuse_monitoring_test_bucket"
+	portNonHNSRun = 9191
+	portHNSRun    = 9192
+)
+
+var prometheusPort = portNonHNSRun
+
+func isPortOpen(port int) bool {
+	c := exec.Command("lsof", "-t", fmt.Sprintf("-i:%d", port))
+	output, _ := c.CombinedOutput()
+	return len(output) == 0
+}
+
+type PromTest struct {
+	suite.Suite
+	// Path to the gcsfuse binary.
+	gcsfusePath string
+
+	// A temporary directory into which a file system may be mounted. Removed in
+	// TearDown.
+	mountPoint string
+
+	enableOTEL bool
+}
+
+// isHNSTestRun returns true if the bucket is an HNS bucket.
+func isHNSTestRun(t *testing.T) bool {
+	storageClient, err := client.CreateStorageClient(context.Background())
+	require.NoError(t, err, "error while creating storage client")
+	defer storageClient.Close()
+	return setup.IsHierarchicalBucket(context.Background(), storageClient)
+}
+
+func (testSuite *PromTest) SetupSuite() {
+	setup.IgnoreTestIfIntegrationTestFlagIsSet(testSuite.T())
+	if isHNSTestRun(testSuite.T()) {
+		// sets different Prometheus ports for HNS and non-HNS presubmit runs.
+		// This ensures that there is no port contention if both HNS and non-HNS test runs are happening simultaneously.
+		prometheusPort = portHNSRun
+	}
+
+	err := setup.SetUpTestDir()
+	require.NoErrorf(testSuite.T(), err, "error while building GCSFuse: %p", err)
+}
+
+func (testSuite *PromTest) SetupTest() {
+	var err error
+	testSuite.gcsfusePath = setup.BinFile()
```

Comment:
Can we use [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/setup/setup.go#L157) for setup?

Reply:
Done